Editorial review: Add information and example for counters in generated content alt text (#42645)
mdn/content: files/en-us/web/css/reference/properties/content/index.md

If using a screen reader, it should speak the word "MOZILLA" when it reaches the image. You can select the `::before` pseudo-element with your developer tools selection tool, and view the {{glossary("accessible name")}} in the accessibility panel.

### Including counters in alternative text

> This example includes a list of links to a set of book chapters, and specifies generated content before each one that includes a book icon and a counter, with alternative text that includes the literal word "Chapter" in place of the icon. This results in an elegant chapter number announcement for screenreader users before each link text is read out.

Reviewer: ```suggestion
This example demonstrates using generated content to include a book icon and a counter, with alternative text that includes the literal word "Chapter" in place of the icon, within links to book chapters. This results in the word "chapter" and chapter number, preceding the text in the link's {{glossary("accessible name")}}, which will be announced to screen reader users when the link receives focus.
```
a bit more pedantic

  <li><a href="#">Three owls</a></li>
  <li><a href="#">No-one answered the door</a></li>
  <li><a href="#">The stranger leaves</a></li>
  <li><a href="#">Bedtime</a></li>
</ol>
```

#### CSS

The CSS includes a {{cssxref("counter-reset")}} for the `chapter` counter on the `<ol>` element. We also increment the `chapter` counter on each `<li>` element using {{cssxref("counter-increment")}}.

```css live-sample___alt-counter
ol {
  counter-reset: chapter;
}

li {
  counter-increment: chapter;
}
```

Next, we set the `<li>` elements' {{cssxref("::marker")}} pseudo-elements to have generated `content` equal to the current `chapter` counter value (retrieved using the {{cssxref("counter()")}} function) plus a colon. We then set the `<a>` elements' {{cssxref("::before")}} pseudo-elements to have generated `content` equal to an empty string (so nothing is displayed), but with alt text equal to the current `chapter` counter value preceded by the word "Chapter".

```css live-sample___alt-counter
li::marker {
  content: counter(chapter) ": ";
}

a::before {
  content: "" / "Chapter " counter(chapter);
}
```

#### Result

{{EmbedLiveSample('alt-counter', '100%', 220)}}

> When using a screenreader, supporting browsers should speak the word "Chapter" followed by the current counter number, followed by the link text, when each list item is reached. So for example "Chapter 1 A stranger calls", "Chapter 2 Two owls", etc. This provides a better experience for screenreader user navigating the list.

Reviewer: ```suggestion
When a screenreader navigates to a link within the list, supporting browsers will announce "Chapter" followed by the current counter number, followed by the link text; such as "Chapter 1 A stranger calls" and "Chapter 2 Two owls".
```
